### DISCUSSION TITLE
jinja template not rendered

### DISCUSSION THREAD

**Discussion by imfaridz** [apache/airflow#37130] [Q&A] [answered]
Hi All,

i currently find issue when using airflow jinja template to be rendered on DataformCreateCompilationResultOperator.
below is the snippets of the code

```
DataformCreateCompilationResultOperator(
            task_id="create_compilation_result",
            project_id=PROJECT_ID,
            region=REGION,
            repository_id=REPOSITORY_ID,
            compilation_result={
                 "git_commitish": GIT_COMMITISH,
                 "code_compilation_config": {
                              "vars": {
                                  "dstart": "{{ prev_execution_date }}",
                                  "dend": "{{ execution_date }}"
                              }
                            }
            },
            gcp_conn_id="gcloud_conn",
  )
```

somehow it is not rendering the {{ prev_execution_date }} and {{ execution_date }} jinja template when it is being executed. 
here's the output of the xcom
```
{'name': 'xxx', 'git_commitish': 'xxx', 'code_compilation_config': {'default_database': 'xxx', 'default_schema': 'xxx', 'assertion_schema': 'xxx', 'vars': {'dstart': '{{ prev_execution_date }}, 'dend': '{{ execution_date }}' }, 'default_location': 'xxxx', 'database_suffix': '', 'schema_suffix': '', 'table_prefix': ''}, 'dataform_core_version': '2.0.1', 'compilation_errors': []}
```

the version of `apache-airflow-providers-google` that i used is 8.7.0
and the `airflow` version is 2.5.1

Much appreciated if anyone can help me on this. thanks!

**Comment by andresitorusz** [comment, marked as answer]
It's not getting rendered by the Airflow because the field `compilation_result` is not in the template_fields of  **DataformCreateCompilationResultOperator**. If you wanna send a templated string or anything to a field in an operator, it has to be in the `template_fields` of the operator.

In [apache-airflow-providers-google==8.7.0](https://airflow.apache.org/docs/apache-airflow-providers-google/8.7.0/_api/airflow/providers/google/cloud/operators/dataform/index.html#airflow.providers.google.cloud.operators.dataform.DataformCreateCompilationResultOperator), the field `compilation_result` is not in the `template_fields`, even the operator doesn't have any `template_fields` in this version. That's why your Jinja template is not getting rendered by the operator.

But, in  [apache-airflow-providers-google==10.14.0](https://airflow.apache.org/docs/apache-airflow-providers-google/stable/_modules/airflow/providers/google/cloud/operators/dataform.html#DataformCreateCompilationResultOperator), you can see the `compilation_result` is already in the `template_fields`, so you can send the templated string to the operator and it will render it.

The closest version of apache-airflow-providers-google to [8.7.0](https://airflow.apache.org/docs/apache-airflow-providers-google/8.7.0/_api/airflow/providers/google/cloud/operators/dataform/index.html#airflow.providers.google.cloud.operators.dataform.DataformCreateCompilationResultOperator) that already use `compilation_result` in the `template_fields` is [10.9.0](https://airflow.apache.org/docs/apache-airflow-providers-google/10.9.0/_api/airflow/providers/google/cloud/operators/dataform/index.html#airflow.providers.google.cloud.operators.dataform.DataformCreateCompilationResultOperator).

![image](https://github.com/apache/airflow/assets/36465150/eaf5bef8-8b9b-4873-9552-66d5b1411d72)

**Reply by andresitorusz** [reply]
Read more about jinja templating in Airflow in this [documentation](https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/operators.html#jinja-templating).

**Reply by andresitorusz** [reply]
@imfaridz you need to upgrade your `apache-airflow-providers-google` version to at least `10.9.0`, if you wanna use the jinja template in `compilation_result` field or parameter.

**Reply by imfaridz** [reply]
Thanks @enchant3dmango , it solves the problem

**Reply by andresitorusz** [reply]
@imfaridz glad to know that.
